script.js

```diff
@@ -1,66 +1,462 @@
+	removeimage.addEventListener("click", showDeletionModal);
+
+	storeLocal();
+
+} //end of additem function
+
+
+
+function deleteItem(e) {
+	_("item" + e.target.dataset.id).remove();
+	_('removeModal').classList.remove("emptyModalParent");
+	_("input_notes").focus();
+	list = list.filter(object => object.id !== e.target.dataset.id);
+	storeLocal();
+	if (list.length === 0) {
+		let emptyImage = "<figure id='emptyImage'><img src='./images/clipboard.svg'><figcaption>Nothing here. Add tasks to view here.</figcaption></figure>";
+		_("list_items").insertAdjacentHTML('afterbegin', emptyImage);
+		_("list_items").classList.add("addEmptyImage");
+		_("searchText").classList.add("noshow");
+	}
 }
 
-//focus on load and submission of input on enter
-window.addEventListener("load", function () {
-  document.getElementById("input_notes").focus();
-  document.getElementById("input_notes").addEventListener("keydown", function (e) {
-    if (e.key === "Enter") {
-      list_additem();
-    }
-  });
-});
+//focus on load and submission of input on enter And adding event to modal button and parent
+window.addEventListener("load", function (e) {
+	_("input_area").addEventListener("submit", e => {
+		_("input_notes").focus();
+		e.preventDefault();
+		list_additem();
+		_("floatDiv").classList.remove("floatDiv");
+		_("floatButton").classList.remove("noshow");
+		_("input_area").classList.remove("flexDisplay");
+	});
+	_("input_notes").focus();
+
+	let closeButton = _('close');
+	let removeModalParent = _('removeModal')
+	let confirmButton = _('confirm');
+	_('popup').addEventListener("click", (e) => { e.stopPropagation(); })
+	confirmButton.addEventListener("click", deleteItem);
+
+	closeButton.addEventListener("click", function () {
+		removeModalParent.classList.remove("emptyModalParent");
+		_("input_notes").focus();
+	})
+
+	removeModalParent.addEventListener("click", function (e) {
+		removeModalParent.classList.remove("emptyModalParent");
+		_("input_notes").focus();
+	});
+	_("searchText").classList.add("noshow");
+	//getting the list from local storage 
+	let listReceived = JSON.parse(localStorage.getItem("listSaved"));
+	if (!listReceived) {
+		listReceived = [];
+	}
+	if (listReceived.length === 0) {
+		let emptyImage = "<figure id='emptyImage'><img src='./images/clipboard.svg'><figcaption>Nothing here. Add tasks to view here.</figcaption></figure>";
+		_("list_items").insertAdjacentHTML('afterbegin', emptyImage);
+		_("list_items").classList.add("addEmptyImage");
+	}
+	if (listReceived) {
+		for (let i = 0; i < listReceived.length; i++) {
+			_("searchText").classList.remove("noshow");
+			list_additem(listReceived[i]);
+		}
+	}
+
+	//error message on closing tab while editing
+	window.onbeforeunload = () => {
+		let confirmationMessage = 'Are you sure you want to leave?';
+		if ((_("input_notes").value.trim() !== "") || list.some((el) => { return el.editMode })) {
+			return confirmationMessage;
+		}
+	}
+	_("x").classList.add("noshow");
```

Comment:
What is `x`? Give proper descriptive IDs to elements.

Reply:
Named it "clear-search"